recipes/gmp/all/conanfile.py

```diff
@@ -100,37 +96,37 @@ def generate(self):
         tc = AutotoolsToolchain(self)
         yes_no = lambda v: "yes" if v else "no"
         tc.configure_args.extend([
-            "--with-pic={}".format(yes_no(self.options.get_safe("fPIC", True))),
-            "--enable-assembly={}".format(yes_no(not self.options.get_safe("disable_assembly", False))),
-            "--enable-fat={}".format(yes_no(self.options.get_safe("enable_fat", False))),
-            "--enable-cxx={}".format(yes_no(self.options.enable_cxx)),
-            "--srcdir={}".format(self.source_folder.replace("\\", "/")),
+            f'--with-pic={yes_no(self.options.get_safe("fPIC", True))}',
+            f'--enable-assembly={yes_no(not self.options.get_safe("disable_assembly", False))}',
+            f'--enable-fat={yes_no(self.options.get_safe("enable_fat", False))}',
+            f'--enable-cxx={yes_no(self.options.enable_cxx)}',
+            f'--srcdir={"../src"}', # Use relative path to avoid issues with #include "$srcdir/gmp-h.in" on Windows
```

Comment:
Shouldn't be necessary, it works fine when backlashs are replaced by forward slashs like it's already done.

Reply:
If I recall correctly, the issue I saw relates to the use of mys2 and the handling of absolute paths with drive letters in them

Reply:
It worked fine in latest PR and msvc build is executed in msys2.

Reply:
Hi @Spacelm,

Here's the error I get without this change:

> checking size of mp_limb_t... 0
> configure: error: Oops, mp_limb_t doesn't seem to work
> ERROR: conanfile.py (gmp/6.2.1): Error in build() method, line 144
>         autotools.configure()
>         ConanException: Error 1 while executing

And here's the corresponding section of build-release/config.log:

> configure:27397: checking size of mp_limb_t
> configure:27402: /c/users/marc/appdata/local/temp/tmp.qzolonp9gf/gmp/src/compile cl -nologo -o conftest.exe  -MD -O2 -Ob2  -DNDEBUG  conftest.c  >&5
> conftest.c
> conftest.c(67): fatal error C1083: Cannot open include file: '/c/users/marc/appdata/local/temp/tmp.qzolonp9gf/gmp/src/gmp-h.in': No such file or directory

The issue being that CL does not know how to open a file with a path specified using the MSYS2 /C/... drive letter syntax. I don't know why CCI is not encountering the same failure mode if it is also using MSYS2 and CL. Maybe it has msys2 configured differently. I am using the version that comes with Git For Windows in an out-of-the-box configuration (no special settings other than the one referred to in https://github.com/conan-io/conan/issues/12760). Thanks.